### DISCUSSION TITLE
Confuso com os padroes de projetos.

### DISCUSSION THREAD

**Discussion by etaptecnologia** [balta-io/forum#170] [Tira Dúvidas] [answered]
Boa tarde @andrebaltieri 

Eu realmente venho gostado muito das suas aulas, de fato você sabe muito sobre o assunto que está lecionando, porem tenho algumas resalvas.

Eu confesso que as aulas estao me gerando muita confusao, principalmente a de dominios ricos, nas aulas de dapper voce mostra um padrao que aparentemente é funcional, mas ele não casa de maneira alguma com o padrão do modulo dominios ricos. isso gera muita confusão.

Outra coisa que vem trazendo bastante confusão é que, o ensino vem sendo passado como se já tivessemos o conhecimento sobre o assunto, e na verdade, é tudo muito novo para a maioria de nós.

Existe algum outro conteudo complementar das aulas de dominios ricos? elas geram um pouco de confusao em algumas coisas.

Obrigado por todo o conteudo maravilhoso que voce esta lecionando.

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @etaptecnologia, como vai?

Eu entendo que possa haver uma certa confusão, até por que não existe um padrão único. Não tem um único caminho, então é normal ter esse sentimento de "será que isto é o mais correto". Principalmente sobre curso avançados como DDD, CQRS e afins.

Sobre a afirmação "tudo muito novo para a maioria de nós", acredito que o mais correto seria "tudo é muito novo para mim", visto que a afirmação é uma visão sua, então vou tomar isto como base para resposta, OK?

É falsa a visão que carga horária dos cursos é a quantidade de horas obrigatórias para seus estudos. Na verdade elas estão muito distantes. Um profissional bem iniciante tem em média 2 anos de estudos, isto com bastante dedicação.

Então, se contarmos toda a carreira .NET (E só ela), tem basicamente 1 ano de estudo e dedicação ali. É necessário persistência, dedicação e muita paciência.

Eu sei que as vezes queremos que as coisas aconteçam rápido, mas não é assim, eu mesmo estou aqui, estudando e já faço isto desde 1998 👴. Nunca paramos!

Desta forma, assistindo os cursos, fazendo as aulas, refazendo as aulas, exercitando o que você aprendeu em outros projetos, tudo isto junto, vai te dar uma base. É um conjunto de inúmeras horas por curso, não basta só assistir.

Sobre o material complementar, é sim sempre bem-vindo. Se está assistindo curso de OOP, ler um livro, estudar mais, se aprofundar mais é sempre bem-vindo 💜

Então, em resumo, sua paciência e persistência que vão demandar todo seu futuro na programação. Continua estudando, com calma, sem pular nada (Vi sua dúvida sobre Key por exemplo, estava na aula, provavelmente você fez na correria e não se atentou).

Continue estudando que vai dar tudo certo! Se tiver qualquer dúvida, estamos aqui 🚀

**Reply by etaptecnologia** [reply]
Boa noite @andrebaltieri .

Realmente tem toda razão.

Mas sobre assistir, eu assisto todas e reviso com muita atencão, acho que seria legal na aula de Dapper falar sobre [key ja que se encontra no codigo no github referente ao modulo
